fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneOnlineIndexingTest.java

```diff
@@ -1344,5 +1351,118 @@ void luceneOnlineIndexingTestTerribleRebalance2ndChance() {
+                    .setContext(context)
+                    .setMetaDataProvider(recordStore.getMetaDataProvider());
+
+            final FDBRecordStore store = builder.createOrOpen(FDBRecordStoreBase.StoreExistenceCheck.NONE);
+
+            // Obtain lock
+            final AgilityContext agile1 = AgilityContext.agile(context, TimeUnit.SECONDS.toMillis(4), 100_000);
+            final FDBDirectory directory = new FDBDirectory(store.indexSubspace(index), index.getOptions(),
+                    null, null, true, agile1);
+
+            final Lock lock1 = directory.obtainLock(WRITE_LOCK_NAME); // will also flush lock
+            agile1.accept(aContext -> {
+                byte [] conflictKey = path.toSubspace(aContext).pack(conflictTuple);
+                final Transaction tr = aContext.ensureActive();
+                tr.get(conflictKey).thenApply(val -> {
+                    say("main thread get: " + Arrays.toString(val));
+                    return null;
+                }).join();
+                tr.set(conflictKey, Tuple.from(100, 20).pack());
+                say("main thread set agile:    " + Arrays.toString(conflictKey));
+            });
+
+
+            // Cause conflict, and wait for it
+            s1.release();
+            acquire(s2);
+            s2.release();
+
+            // finalize
+            boolean gotException = false;
+            try {
+                say("agile will commit now");
+                try {
+                    lock1.close();
+                } catch (RuntimeException ex) {
+                    gotException = true;
+                }
+                try {
+                    directory.close();
+                } catch (RuntimeException ex) {
+                    gotException = true;
+                }
+                agile1.flushAndClose();
+                context.commit();
+            } catch (RuntimeException ex) {
+                gotException = true;
+            } finally {
+                assertTrue(gotException);
+            }
+        }
+        conflictThread.join();
+
+        // ensure that a new lock is possible
+        TestRecordsTextProto.SimpleDocument doc3 = createSimpleDocument(1623L, "Salesmen jokes are funnier", 2);
+        try (final FDBRecordContext context = fdb.openContext()) {
+            FDBRecordStore.Builder builder = recordStore.asBuilder()
+                    .setContext(context)
+                    .setMetaDataProvider(recordStore.getMetaDataProvider());
+
+            final FDBRecordStore store = builder.createOrOpen(FDBRecordStoreBase.StoreExistenceCheck.NONE);
+            store.saveRecord(doc3);
+            context.commit();
+        }
+    }
+
+    private void say(String msg) {
+        LOGGER.debug("::::::" + msg);
+    }
+
+    private void acquire(Semaphore semaphore) {
+        try {
+            semaphore.acquire();
+        } catch (InterruptedException ex) {
+            throw new RuntimeException(ex);
```

Comment:
Not super important because it's a test, but generally, you should always re-interrupt the current thread if you catch `InterruptedException`, and it's good to be consistently in that habit.